audio/writer.h

```diff
@@ -0,0 +1,26 @@
+#ifndef AUDIO_WRITER_H
+#define AUDIO_WRITER_H
+
+#include <stdio.h>
+#include <soundio/soundio.h>
+
+struct audio_writer {
+	void (*write)(struct audio_writer *writer, enum SoundIoFormat fmt,
+		      char *data, int size, int bytes_per_frame);
```

Comment:
Should we `typedef` these function pointers? I'm not normally a fan of `typedef`, but I do find it makes function pointers easier to work with in all contexts.